### DISCUSSION TITLE
Checks unification for MagickGeometry

### DISCUSSION THREAD

**Discussion by Gounlaf** [dlemstra/Magick.NET#1568] [General]
:wave:

`MagickGeometry` has internal check for dimensions (Percentage of absolute)

https://github.com/dlemstra/Magick.NET/blob/708c592f9a1bd835b81f5131beee4ca0fcca8c09/src/Magick.NET/Types/MagickGeometry.cs#L50-L55

https://github.com/dlemstra/Magick.NET/blob/708c592f9a1bd835b81f5131beee4ca0fcca8c09/src/Magick.NET/Types/MagickGeometry.cs#L322-L325

On some method (in `MagickImage` as example), same check has `MagickGeometry` is done:
https://github.com/dlemstra/Magick.NET/blob/708c592f9a1bd835b81f5131beee4ca0fcca8c09/src/Magick.NET/MagickImage.cs#L3474-L3479

On some others, it's not:
https://github.com/dlemstra/Magick.NET/blob/708c592f9a1bd835b81f5131beee4ca0fcca8c09/src/Magick.NET/MagickImage.cs#L3894-L3897

What do you prefer? Double check or reducing redundant calls?

I will do a PR to unify all checks depending on your preference.

Regards.

**Comment by dlemstra** [comment]
We can remove the redundant checks. But we might need some extra unit tests to make sure exceptions are still raised?

**Reply by Gounlaf** [reply]
All existing tests will be kept

**Reply by dlemstra** [reply]
You might need to add extra ones if I forgot to add them.

**Reply by Gounlaf** [reply]
Oh sure; that's not a problem 🙂

Will do a Draft PR soon and push several commits to make review easier.

**Comment by Gounlaf** [comment]
Note: `Percentage` class doesn't have checks like `MagickGeometry`. However, comments suggest it should have. It should have checks too, and, as well, redundant checks could be removed.

If you are ok with this too, I will do another PR, the current one focusing on `MagickGeometry`

**Reply by dlemstra** [reply]
I don't think we need negative percentages so it is probably a good idea to add checks for this.
